src/main/scala/org/dia/algorithms/mcc/NetcdfDFSMCC.scala

```diff
@@ -152,157 +131,78 @@ object NetcdfDFSMCC {
-              min = currentProperties.get._3
-              if (value < min) min = value
-              if (value > max) max = value
-            } else {
-              min = value
-              max = value
-            }
-            area += 1
-            areaMinMaxTable += ((frame + ":" + label, (area, max, min)))
+            val node = nodeMap.getOrElse(frame + ":" + label, new MCCNode(frame.toInt, label))
+            node.update(value, row, col)
+            nodeMap.update((frame + ":" + label), node)
           }
         }
 
+
         for (row <- 0 to product.rows - 1) {
           for (col <- 0 to product.cols - 1) {
             /** Find non-zero points in product array */
+            updateComponent(components1(row, col), t1.metaData("FRAME"), t1.tensor(row, col), row, col)
+            updateComponent(components2(row, col), t2.metaData("FRAME"), t2.tensor(row, col), row, col)
             if (product(row, col) != 0.0) {
-              /** save components ids */
-              val value1 = components1(row, col)
-              val value2 = components2(row, col)
-              overlappedPairsList += ((value1, value2))
+
+              /** If overlap exists create and edge and update overlapped area */
+              val frame1 = t1.metaData("FRAME")
+              val label1 = components1(row, col)
+              val node1 = nodeMap(frame1 + ":" + label1)
+
+              val frame2 = t2.metaData("FRAME")
+              val label2 = components2(row, col)
+              val node2 = nodeMap(frame2 + ":" + label2)
+
+              val edgeKey = s"$frame1:$label1,$frame2:$label2"
+              val edge = if (MCCEdgeMap.contains(edgeKey)) MCCEdgeMap(edgeKey) else new MCCEdge(node1, node2)
+              edge.incrementAreaOverlap()
+              MCCEdgeMap.update(edgeKey, edge)
             }
-            updateComponent(components1(row, col), t1.metaData("FRAME"), t1.tensor(row, col))
-            updateComponent(components2(row, col), t2.metaData("FRAME"), t2.tensor(row, col))
           }
         }
 
-        /**
-         * This code essentially computes the number of times the same tuple occurred in the list,
-         * a repeat occurance would indicate that the components overlapped for more than one cell
-         * in the product matrix. By calculating the number of overlaps we can calculate the number of cells
-         * they overlapped for and since each cell is of a fixed area we can compute the area overlap
-         * between those two components.
-         */
-        var overlappedMap = overlappedPairsList.groupBy(identity).mapValues(_.size)
-        println(s"Overlap Map ${overlappedMap.size}")
+        val filtered = MCCEdgeMap.filter({
+          case (k, edge) =>
+            val srcNode = edge.srcNode
+            val (srcArea, srcMinTemp, srcMaxTemp) = (srcNode.area, srcNode.minTemp, srcNode.maxTemp)
+            val isSrcNodeACloud = (srcArea >= 2400) || (srcArea < 2400 && (srcMinTemp / srcMaxTemp) < 0.9)
```

Comment:
2400 should be 150 (number of boxes).
Actually, this should be a var that is passed into the function. 
Aslo the value of the convective fraction (0.9) should also be a var.
e.g  nodeMinArea:Int
       convectiveFraction:Double